### PR TITLE
Make the workaround for absolute paths in depfiles compatible with macOS

### PR DESCRIPTION
**Pull request by jason-simmons**
See https://github.com/flutter/engine/pull/49542

### CODE REVIEW

**Review by jmagman — APPROVED** [review]
On master on macOS:
```
/fml/platform/android/fml.scoped_java_ref.o && sed -i 's@/b/f/w@../..@g' obj/flutter/fml/platform/android/fml.scoped_java_ref.o.d
sed: 1: "obj/flutter/fml/platfor ...": invalid command code o
```
On this PR: success.

File: build/toolchain/gcc_toolchain.gni [hunk 1]
```diff
@@ -122,14 +122,20 @@ template("gcc_toolchain") {
       coverage_flags = "-fprofile-instr-generate -fcoverage-mapping"
     }
 
+    if (host_os == "mac") {
```
Comment (zanderso) [inline build/toolchain/gcc_toolchain.gni:125]: `-i''` that is, an empty string with no space after the `-i` should work on both, but maybe it's better to be explicit like this.

**Review by zanderso — APPROVED** [review]

File: build/toolchain/gcc_toolchain.gni (hunk 1 above)
Comment (jason-simmons) [reply build/toolchain/gcc_toolchain.gni:125]: In my local testing on macOS `-i''` did not work.

It looked like it was being parsed as equivalent to `-i`, causing the next argument (the  sed command) to be used as the backup file extension.

Comment (zanderso) [reply build/toolchain/gcc_toolchain.gni:125]: Ah, maybe it only works with `-e` before the command to delineate the empty string from the command like: `sed -i'' -e 's@/b/f/w@../..@g' $depfile`.

Comment (jason-simmons) [reply build/toolchain/gcc_toolchain.gni:125]: I tried that originally but found that it was creating a backup file with the extension `-e` added to the filename

Comment (zanderso) [reply build/toolchain/gcc_toolchain.gni:125]: Hahaha. Okay. I'm glad you checked =)

**Comment by jmagman** [comment]
New errors in roll:
```
ERROR at //build/toolchain/gcc_toolchain.gni:128:27: Assignment had no effect.
      sed_in_place_flag = "-i"
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8759705668090838113/+/u/gn_--web_--runtime-mode_release_--no-goma_--no-lto/stdout

https://github.com/flutter/engine/pull/49542

**Comment by jason-simmons** [comment]
Fix for the unused variable error: https://github.com/flutter/buildroot/pull/810

**Pull request merged**